Marlin/src/sd/cardreader.cpp

```diff
@@ -352,6 +353,9 @@ void CardReader::initsd() {
   else {
     flag.detected = true;
     SERIAL_ECHO_MSG(MSG_SD_CARD_OK);
+    #if ENABLED(EEPROM_SETTINGS) && DISABLED(FLASH_EEPROM_EMULATION)
+      (void)settings.load();
+    #endif
```

Comment:
Apparently the ESP32 platform uses SPIFFS to talk to the onboard SD card, and supports any arbitrary external SD card through the SDFat interface. So this linkage / side-effect might cause a problem on that platform.

Of course, this currently causes problems on every platform (including AVR) as it stands, because every SD card initialization will cause the EEPROM to be reloaded. So the conditions need a-fixing, at the very least!